### PR TITLE
2019 11 30 key manager

### PR DESCRIPTION
**Pull request by Christewart**
This PR segregates key handling/storage logic from the wallet project. For the most part, this is a move only PR taking things inside the `wallet` project and moving them to the new `key-manager` project.

For now, this just delegates storage of the encrypted mnemonic seed to the key manager project. Over the log run, we want to make it so that the wallet project needs to communicate with the key-manager to access private keys. 

This means that ALL SIGNING should be done inside of the key-manager, and private keys should not be leaked outside of the key manager. This makes it easier to reason about the security characteristics of our private keys, and a way to provide a uniform interface for alternative key storage systems (hsm, cloud based key storage, etc) to be plugged into the bitcoin-s library. 

##### TODO 
- [x] key-manager.md
- [x] duplicate encrypted seed unit test

### CODE REVIEW

File: docs/applications/key-manager.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+---
+
+
+#### Key Manager
+
+The key manager module's goal is to encapusulate all private key interactions with the [wallet](wallet.md) project.
+
+As of this writing, the wallet just delegates storage of the encrypted mnemonic seed to the key manager project. Over the log run, we want to make it so that the wallet project needs to communicate with the key-manager to access private keys.
+
+This means that ALL SIGNING should be done inside of the key-manager, and private keys should not be leaked outside of the key manager.
```
Comment (nkohen) [inline docs/applications/key-manager.md]: nit "should not be leaked outside of" -> "should not leave" (they shouldn't be leaked anywhere ;)

File: testkit/src/main/scala/org/bitcoins/testkit/fixtures/BitcoinSFixture.scala [hunk 2]
```diff
@@ -36,6 +36,9 @@ trait BitcoinSFixture extends BitcoinSAsyncFixtureTest {
       test(fixture.asInstanceOf[FixtureParam]).toFuture
     }
 
+    outcomeF.failed.foreach(err =>
+      println(s"Failed fixture test execution=${err}"))
```
Comment (nkohen) [inline testkit/src/main/scala/org/bitcoins/testkit/fixtures/BitcoinSFixture.scala:40]: nit: `println -> logger.error`

File: wallet/src/main/scala/org/bitcoins/wallet/config/WalletAppConfig.scala [hunk 3]
```diff
@@ -73,6 +72,16 @@ case class WalletAppConfig(
     initF
   }
 
+  /** The path to our encrypted mnemonic seed */
+  protected[wallet] def seedPath: Path = {
```
Comment (nkohen) [inline wallet/src/main/scala/org/bitcoins/wallet/config/WalletAppConfig.scala]: can't this be `private[wallet]`? case classes can't be extended and I think this grants more access than we want (protected is weird)

File: key-manager-test/src/test/scala/org/bitcoins/keymanager/WalletStorageTest.scala [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    assert(walletConf.seedExists())
+
+    val exn: RuntimeException = intercept[RuntimeException] {
+      //attempt to write another mnemonic
+      getAndWriteMnemonic(walletConf)
+    }
+
+    //need this since we are using async testing suite
+    //and exn is not a Future
+    succeed
```
Comment (rorp) [inline key-manager-test/src/test/scala/org/bitcoins/keymanager/WalletStorageTest.scala]: `assertThrows()` returns an `Assertion`

File: key-manager-test/src/test/scala/org/bitcoins/keymanager/WalletStorageTest.scala [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      WalletStorage.decryptMnemonicFromDisk(seedPath,passphrase)
+    read match {
+      case ReadMnemonicSuccess(readMnemonic) =>
+        assert(writtenMnemonic == readMnemonic)
+      case err: ReadMnemonicError => fail(err.toString)
+    }
+  }
+
+  it must "fail to read a mnemonic with bad password" in { walletConf =>
+    val writtenMnemonic = getAndWriteMnemonic(walletConf)
```
Comment (benthecarman) [inline key-manager-test/src/test/scala/org/bitcoins/keymanager/WalletStorageTest.scala]: This val is never used

**Comment by benthecarman** [comment]
CI is failing for almost all the cases

File: testkit/src/main/scala/org/bitcoins/testkit/fixtures/BitcoinSFixture.scala (hunk 2 above)
Comment (Christewart) [reply testkit/src/main/scala/org/bitcoins/testkit/fixtures/BitcoinSFixture.scala:40]: I'm going to keep this now to help with debugging on CI as we have logging turned off there

**Pull request merged**